lib/modules/dosomething/dosomething_signup/dosomething_signup.module

```diff
@@ -581,8 +581,33 @@ function dosomething_signup_user_signup($nid, $account = NULL, $source = NULL) {
       return;
     }
 
-    // Set default signup message if we're still in this function.
-    dosomething_signup_set_takeover();
+    $existing_user = dosomething_signup_not_first_signup($account->uid);
+
+    // Set signup message, depending on type of user with specific params.
+    if ($existing_user) {
+      // pass repeat-signup-existing in source param to
+      dosomething_signup_set_signup_message();
+    }
+    else {
+      // pass repeat-signup-new in source param to
+      dosomething_signup_set_takeover();
+    }
+  }
+}
+
+/**
+ * Returns true if user this is not a user's first signup.
+ * @param int $user uid
+ */
+function dosomething_signup_not_first_signup($uid) {
```

Comment:
that makes total sense! I was even getting a little confused when writing it / had to double check it a ton. I'll change that now!